app/src/main/java/org/mozilla/fenix/utils/OnVisualCompletenessExecutionQueue.kt

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import android.view.View
+import kotlinx.coroutines.Job
+import mozilla.components.support.base.log.Log
+
+/**
+ * A queue of paused `Job`s whose execution is resumed when Fenix is visually complete.
+ *
+ * This queue contains paused `Job`s. Each Job in the queue will be restarted once Fenix
+ * is visually complete (as long as it has not be cancelled since it was added to the queue).
+ * Each Job will execute on the thread on which it was originally scheduled.
+ *
+ * This is threadsafe.
+ */
+object OnVisualCompletenessExecutionQueue {
+    internal var jobs = mutableListOf<Job>()
```

Comment:
At a high level, what we need is to execute tasks at a given point in time. We're currently using `Job` to do that but `Job` is more complex than we need, i.e. "a task executing at a given point in time": Jobs force you to use coroutines, support cancellation, etc. and that's prevalent by all the extra if statement checks below and the need for unit tests. Can we use something simpler?